scripts/pool-setup.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint @typescript-eslint/no-var-requires: "off" */
+import { createSubstrateAPI } from "../src/factory";
+import { ApiPromise } from "@polkadot/api";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+import { addHexPrefix, newAccountId, stripHexPrefix } from "../src";
+
+const yargs = require("yargs/yargs");
+const { hideBin } = require("yargs/helpers");
+const args = yargs(hideBin(process.argv))
+    .option("parachain-endpoint", {
+        description: "The wss url of the parachain",
+        type: "string",
+    })
+    .option("x-tokens-parachain-id", {
+        description: "The parachain id of the destination",
+        type: "number",
+        demandOption: true,
+    })
+    .option("x-tokens-currency-id", {
+        description: "The currency to send",
+        type: "string",
+        demandOption: true,
+    })
+    .option("x-tokens-amount", {
+        description: "The amount to send",
+        type: "number",
+        demandOption: true,
+    })
+    .option("destination-address", {
+        description: "The hex-encoded account id",
+        type: "string",
+    })
+    .option("destination-sovereign", {
+        description: "Indicates we should transfer to the sovereign account",
+        type: "bool",
+    })
+    .option("with-defaults-of", {
+        description: "Which default values to use",
+        choices: ['kintsugi', 'interlay'],
+    })
+    .argv;
+
+main().catch((err) => {
+    console.log("Error thrown by script:");
+    console.log(err);
+});
+
+function printExtrinsic(extrinsic: SubmittableExtrinsic<"promise">, endpoint: string) {
+    console.log("Data:", extrinsic.method.toHex());
+    console.log("Hash:", extrinsic.method.hash.toHex());
+    const url =
+        "https://polkadot.js.org/apps/?rpc=" +
+        encodeURIComponent(endpoint) +
+        "#/extrinsics/decode/" +
+        extrinsic.method.toHex();
+    console.log("Url:", url);
+    console.log("");
+}
+
+function palletIdToAccountId(api: ApiPromise, palletId: any): string {
+    const accountId = Buffer.concat([
```

Comment:
Nit: This could be condensed to:
```
Buffer.concat([
    Buffer.from("modl"),
    Buffer.from(palletId)
  ],
  32
);
```
ie. we can skip the calculation with magic numbers.

Reply:
Example test case:
```
    it.only("buffer concat", () => {
        const abc = Buffer.from("abc");
        const one23 = Buffer.from("123");
        const buf = Buffer.concat([abc, one23], 32);

        // kept length
        expect(buf.length).to.be.eq(32);
        // filled overallocated bytes with zeroes
        expect(buf.readBigUInt64BE(abc.length + one23.length)).to.be.eq(BigInt(0));
    });
```
^ passes for me

Reply:
Thanks! I wonder if we should actually move this code into the utils and actually add those unit tests?

Reply:
> Thanks! I wonder if we should actually move this code into the utils and actually add those unit tests?

We can move the code to the utils if there is a shared use case now, or in future for that matter. 
I'm not aware if any of the other scripts do something similar.